src/sensai/util/cache_azure.py

```diff
@@ -498,14 +575,8 @@ def _getFromInMemoryDf(self, key):
     def _getPartitionKeyForRowKey(self, key: str):
         return self._batchCommitTable.tableName if self._partitionKeyGenerator is None else self._partitionKeyGenerator(key)
 
-    def _commitDeferred(self):
-
-        def doCommit():
-            self._batchCommitTable.commit(self._maxBatchSize, self._deferredCommitDelaySecs)
+    def _commit(self):
+        self._batchCommitTable.commitNonBlockingCurrentQueueState(self._maxBatchSize)
```

Comment:
The use of this commit function will cause an incomplete commit in this sequence of events (executing thread is indicated by [d] for deferred commit thread, [m] for main):
1. [d] deferred commit starts
2. [d] r = queue.popAll()
3. [m] insert into queue
4. [m] check if [d] is active
5. [m] do not spawn new thread because thread [d] is still alive
6. [d] actual commit of r
6. [d] terminate
i.e. what thread [m] inserted into the queue in step 3 will never be committed, since no new thread is spawned.